js/style/style.js

```diff
@@ -633,6 +633,15 @@ Style.prototype = util.inherit(Evented, {
     },
 
     queryRenderedFeatures: function(queryGeometry, params, classes, zoom, bearing) {
+        if (params.filter) {
+            if (validateStyle.emitErrors(this, validateStyle.filter({
+                key: 'queryRenderedFeatures.filter',
+                value: params.filter,
+                style: this.serialize(),
+                styleSpec: styleSpec
+            }))) return this;
```

Comment:
Yes, the return value should be an empty array.

We don't throw anywhere else in `Style`, even when we can. I think there's value in consistency in error reporting.